pkg/kosmosctl/floater/check.go

```diff
@@ -199,60 +203,84 @@ func (o *CommandCheckOptions) Run() error {
 func (o *CommandCheckOptions) RunRange(iPodInfos []*FloatInfo, jPodInfos []*FloatInfo) []*PrintCheckData {
 	var resultData []*PrintCheckData
 
+	ch := make(chan int, o.routinesMaxNum)
```

Comment:
I do not know why do you add a channel here ？

Reply:
I want the number of concurrent threads to be manageable.
Waitgroup only exposed me to the add, done, wait methods.

Reply:
I prefer to make utils to deal the concurrency

Reply:
sorry，I'm not sure I know what you mean.
My idea is to control the maximum number of concurrent routines by configuring the routines-max-number parameter and channel. This is because Waitgroup cannot control the number of concurrent threads. When the cluster size is very large, waitgroup can run out of resources.
you means I need to find Third-party library to deal. I didn't find a concurrency control method in the official library.